text/0088-fix-memory-limitiations-in-session-replays-access-pattern.md

```diff
@@ -0,0 +1,288 @@
+- Start Date: 2023-04-24
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/88
+- RFC Status: draft
+
+# Summary
+
+The Session Replay index page will run out of memory when processing queries for our largest customers. This document contains proposed solutions for addressing this short-coming.
+
+# Motivation
+
+Our goal is to make the Session Replay product highly scalable. We expect changes to the product will eliminate our OOM issues and increase performance for customers with large amounts of ingested data.
```

Comment:
I didn't see this in the solution options considered if I understand them correctly: But would limiting the number of replays shown on the Index page materially improve the OOM issues or probability of them occurring?

Right now we display 50 replays upon load of this page which is * a lot * and seems unnecessary. I assume the number of replays we display on this view, will result in fetching more info on sort/filter actions and retention period changes as well.

Reply:
> I didn't see this in the solution options considered if I understand them correctly: But would limiting the number of replays shown on the Index page materially improve the OOM issues or probability of them occurring?

Unfortunately no.  Whats displayed and what was actually aggregated in the database are not the same in this case.  The "scan range" (the stats period value: 14 days, 24 hours, 90 days, etc) is what creates these OOM issues.

> Right now we display 50 replays upon load of this page which is * a lot * and seems unnecessary.

If it makes the product better we should do it!  Reducing the row set will have a positive impact on performance.